x-pack/plugins/security_solution/public/timelines/containers/host_risk_score/use_host_risk_score.ts

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { Observable } from 'rxjs';
+import { filter } from 'rxjs/operators';
+
+import { i18n } from '@kbn/i18n';
+
+import { useCallback, useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+
+import { useObservable, withOptionalSignal } from '@kbn/securitysolution-hook-utils';
+
+import { useAppToasts } from '../../../common/hooks/use_app_toasts';
+import { useKibana } from '../../../common/lib/kibana';
+import { inputsActions } from '../../../common/store/actions';
+import { RISKY_HOSTS_INDEX } from '../../../../common/constants';
+import { isIndexNotFoundError } from '../../../common/utils/exceptions';
+import { useIsExperimentalFeatureEnabled } from '../../../common/hooks/use_experimental_features';
+
+import {
+  DataPublicPluginStart,
+  isCompleteResponse,
+  isErrorResponse,
+} from '../../../../../../../src/plugins/data/public';
+import {
+  HostsQueries,
+  HostRiskScoreRequestOptions,
+  HostRiskScoreStrategyResponse,
+} from '../../../../common';
+
+type GetHostRiskScoreProps = HostRiskScoreRequestOptions & {
+  data: DataPublicPluginStart;
+  signal: AbortSignal;
+};
+
+const getHostsRiskScore = ({
+  data,
+  defaultIndex,
+  hostName,
+  signal,
+}: GetHostRiskScoreProps): Observable<HostRiskScoreStrategyResponse> =>
+  data.search.search<HostRiskScoreRequestOptions, HostRiskScoreStrategyResponse>(
+    {
+      defaultIndex,
+      factoryQueryType: HostsQueries.riskScore,
+      hostName,
+    },
+    {
+      strategy: 'securitySolutionSearchStrategy',
+      abortSignal: signal,
+    }
+  );
+
+const getHostRiskScoreComplete = (
+  props: GetHostRiskScoreProps
+): Observable<HostRiskScoreStrategyResponse> => {
+  return getHostsRiskScore(props).pipe(
+    filter((response) => {
+      return isErrorResponse(response) || isCompleteResponse(response);
+    })
+  );
+};
+
+const getHostRiskScoreWithOptionalSignal = withOptionalSignal(getHostRiskScoreComplete);
+
+const useHostRiskScoreComplete = () => useObservable(getHostRiskScoreWithOptionalSignal);
+
+const QUERY_ID = 'host_risk_score';
+const noop = () => {};
+
+export interface HostRiskScore {
+  loading: boolean;
+  isModuleEnabled?: boolean;
+  fields: Array<{ field: string; value: string }>;
+}
+
+export const useHostRiskScore = ({
```

Comment:
There is a lot of overlap between this hook and those introduced in https://github.com/elastic/kibana/pull/109553; I think we need a pass here to make sure we're not duplicating logic and reusing code when possible.

Reply:
Hi @rylnd! Thank you for helping me out with this PR. 

I've just done some cleaning on this hook, to remove logic that isn't necessary.

I started working on this PR by copying and pasting the hook from the previously open PR https://github.com/elastic/kibana/pull/109553 so I wouldn't be blocked by it. But they have diverged.

Indeed they still have a lot in common, but most of the overlapping logic is the same for every hook that fetches data using a `SearchStrategy`. They are all very similar.  
I could identify the following repeated logic that is unique for the Host risk domain:
* `IndexNotFoundError` error handling and `isModuleEnabled` logic
* Both query the same index

Considering this, I think that we have two ways forward. 
1. Abstracting the repeated logic (maybe in a hook) so both can reuse it.
2. Refactor https://github.com/elastic/kibana/pull/109553 hook to make it way more configurable so both views can use it.

What is your opinion?

Good to know about https://github.com/elastic/security-team/issues/1785. Do you already know how is it going to be implemented? Could both hooks reuse this solution?

Reply:
@machadoum my understanding is that the only fundamental difference between the two divergent search strategies + hooks is which filters are applied:

* Overview requires a date filter, but also allows arbitrary filters from the query bar
* Details requires a host name filter

Details also applies some minor transformation on the response, but IMO this could be pushed to the frontend as Overview is currently doing.

My takeaway was that these differences did not warrant two separate implementations, and that it would be relatively simple to delete `useHostRiskScore` and instead use `useRiskyHostsComplete`, but I defer to you and @ecezalp on the best course of action here.

Regarding https://github.com/elastic/security-team/issues/1785, the change should be _very_ minor for both cases: the only thing changing will be the `index` argument, which will now need to include the current `spaceId`. Again, please work with @ecezalp to find a shared/consistent solution across efforts.

Thanks!